sos/archive.py

```diff
@@ -135,6 +135,9 @@ class FileCacheArchive(Archive):
     def __init__(self, name, tmpdir, policy, threads, enc_opts, sysroot,
                  manifest=None):
         self._name = name
+        # truncate the name just relative to the tmpdir in case of full path
+        if os.path.commonprefix([self._name, tmpdir]) == tmpdir:
```

Comment:
This condition is required due to `TarFileArchive.add(`testfile`) where we add files without their path.